openfisca_core/parameters.py

```diff
@@ -559,17 +561,183 @@ def __getattr__(self, key):
         param_name = _compose_name(self._name, key)
         raise ParameterNotFound(param_name, self._instant_str)
 
-    def __getitem__(self, key):  # deprecated
-        return getattr(self, key)
+    def __getitem__(self, key):
+        # If fancy indexing is used, cast to a vectorial node
+        if isinstance(key, np.ndarray):
+            return VectorialParameterNodeAtInstant.build_from_node(self)[key]
+        return self._children[key]
 
     def __iter__(self):
         return iter(self._children)
 
 
+class VectorialParameterNodeAtInstant(object):
+    """
+        Parameter node of the legislation at a given instant which has been vectorized.
+        Vectorized parameters allow requests such as parameters.housing_benefit[zipcode], where zipcode is a vector
+    """
+
+    @staticmethod
```

Comment:
Why is it a `@staticmethod` and not `@classmethod`. Shouldn't it be applied to the new declared class? Or, if it's mainly a way to unite functions, why is `contains_nan` outside of this class? 🤔

Reply:
> Why is it a @staticmethod and not @classmethod

Because it doesn't rely on the class. But it's clearly related to `VectorialParameterNodeAtInstant`, as this method is a builder for this class. However, `contains_nan` is not directly related to `VectorialParameterNodeAtInstant`.